### PR TITLE
throw error for PC and PJ when projectRef is not resolved

### PR DESCRIPTION
**Pull request by zhili1208**
https://github.com/NuGet/Home/issues/5777

For netcore project and package reference, NuGet always pass project reference path to restore, let restore throw error.

But Project.json and packages.config will skip unresolved project reference, let restore succeed, but fail during build.

the fix is let PJ and PC also throw during restore if project reference is not resolved.

we should have test infrastructure for error/warning code, filed a bug: https://github.com/NuGet/Home/issues/5792

### CODE REVIEW

**Comment by emgarten** [comment]
Check where the Resolved property was originally added. I know this was added for a DTE related reason. There could be different behavior for every project system around this.

**Comment by zhili1208** [comment]
the Resolved is added before NuGet.Client repo created, so can't find the original commit, searched all bugs, didn't find anything

**Review by jainaashish — CHANGES_REQUESTED** [review]
Please add test for this scenario. Adding whole infra for warnings n errors will happen when it happens, but that should not stop us adding tests for scenarios which can be easily added.

**Comment by emgarten** [comment]
I'm leery of taking this without finding the original issue. Other things may have changed around this that fix the issue, but I recall a lot of pitfalls around this check.

**Comment by emgarten** [comment]
> Resolved is added before NuGet.Client repo create

That doesn't seem right, but if so the original git history was copied from the other repos.

**Comment by zhili1208** [comment]
this change is introduced by https://github.com/NuGet/NuGet.Client/commit/9880fd22f0c0c1c29405daf280810bb8cab24769 which fixed https://github.com/NuGet/Home/issues/1110,

Here is the PR: https://github.com/NuGetArchive/NuGet.VisualStudioExtension/pull/54

the fix is about native projects where the reference does not exist on disk DTE throws a generic exception.

the "try catch" is the real fix,  Removing Resovled Check doesn't break this

File: test/EndToEnd/tests/BuildIntegratedTest.ps1 [hunk 1]
```diff
@@ -588,4 +588,22 @@ function Test-BuildIntegratedLegacyRebuildDoesNotDeleteCacheFile {
...
+    [SkipTestForVS14()]
+
+    $projectR = New-Project Net45BuildIntegratedClassLibrary
+    $projectT = New-Project PackageReferenceClassLibrary
+
+    $projectR | Add-ProjectReference -ProjectTo $projectT
+
+    Clean-Solution
+
+    $projectT = $projectT | Select-Object UniqueName, ProjectName, FullName
```
Comment (nkolev92) [inline test/EndToEnd/tests/BuildIntegratedTest.ps1:603]: This line is not needed.

File: test/EndToEnd/nuget.assert.ps1 [hunk 2]
```diff
@@ -117,6 +117,31 @@ function Assert-ProjectJsonLockFilePackage {
...
+
+    $lockFile = Get-ProjectJsonLockFile $Project
+
+    Assert-NotNull $lockFile
+
+    $found = $false
+
+    foreach ($log in $lockFile.LogMessages) {
+        
+        if ($log.Code.ToUpperInvariant().Equals($errorCode.ToUpperInvariant()))
```
Comment (nkolev92) [inline test/EndToEnd/nuget.assert.ps1:136]: This would fail, code is an enum, so replace this with ToString()

**Review by nkolev92 — COMMENTED** [review]
Let's discuss this together with @emgarten and @jainaashish at the least. 
I'd personally prefer the other fix, where we let msbuild deal with it.

File: test/EndToEnd/tests/BuildIntegratedTest.ps1 [hunk 3]
```diff
@@ -588,4 +588,22 @@ function Test-BuildIntegratedLegacyRebuildDoesNotDeleteCacheFile {
 
     #Assert
```
Comment (nkolev92) [inline test/EndToEnd/tests/BuildIntegratedTest.ps1:589]: I would also suggest adding an LSL test for the same scenario. 
I did some testing myself, so I used the following test. I would add it to LightweightSolutionLoadTest.ps1

```
function Test-DeferredBuildIntegratedGetIncompatibleError {
    [SkipTestForVS14()]
      
    $projectR = New-Project Net45BuildIntegratedClassLibrary #project.json
    $projectT = New-Project PackageReferenceClassLibrary #this is a net46 library, incompatible

    $projectR | Add-ProjectReference -ProjectTo $projectT

    Clean-Solution

    $projectR = $projectR | Select-Object UniqueName, ProjectName, FullName

    Enable-LightweightSolutionLoad -Reload

    # Act (Restore)
    Build-Solution

    Assert-True ($projectR | Test-Project -IsDeferred) -Message 'Test project should stay in deferred mode'
    
    Assert-ProjectJsonLockFileErrorCode $projectR NU1201
}
```

**Comment by zhili1208** [comment]
after discussion, we decide to close it as won't fix

**Pull request closed without merging**